server/src/main/java/org/apache/uniffle/server/merge/ShuffleEntity.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.server.merge;
+
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.Map;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.roaringbitmap.longlong.Roaring64NavigableMap;
+
+import org.apache.uniffle.common.ShufflePartitionedBlock;
+import org.apache.uniffle.common.ShufflePartitionedData;
+import org.apache.uniffle.common.config.RssConf;
+import org.apache.uniffle.common.util.JavaUtils;
+import org.apache.uniffle.server.ShuffleServer;
+
+public class ShuffleEntity<K, V> {
```

Comment:
Could we give a better name? What's the name of similar concept in the Hadoop?

Reply:
It is difficult to find a similar concept in Hadoop. Perhaps org.apache.hadoop.mapreduce.task.reduce.MergeManagerImpl is the most similar

In this PR, org.apache.uniffle.server.merge.ShuffleMergeManager is the manager for all shuffle which need to merge. ShuffleEntity is a component of ShuffleMergeManager, used to manage shuffles that need to be merged. PartitionEntity is a component of ShuffleEntity, used to manage partitions that need to be merged.

If change name, I will change ShuffleEntity to Shuffle, and PartitionEntity to Partition. Because the two class already in merge package, though the two names are too common, I think Shuffle and Partition is the best name. Do you think so? @jerqi

Reply:
> OK. Shuffle and Partition may better.

OK, I have rename all.